doc/source/cookbook.rst

```diff
@@ -1318,3 +1318,96 @@ of the data values:
        'weight': [100, 140, 180],
        'sex': ['Male', 'Female']})
    df
+
+Categorical Data
+----------------
+
+Using categorical data type to store data more efficiently and consistently in multiple DataFrames
+
+Create two DataFrames with one column each from a subset of the unique values with overlap. Append the dataframes and examine the memory usage.
+
+.. ipython:: python
+
+   from pandas.api.types import CategoricalDtype
+
+   domain = np.array(['A','B','C','D','E','F','G','H','I','J','a','b','c','d','e','f','g','h','i','j'])
+
+   df1 = pd.DataFrame({'A': domain.take(np.random.randint(0,12,size=100000))})
+   df2 = pd.DataFrame({'A': domain.take(np.random.randint(8,20,size=100000))})
+
+   df3 = df1.append(df2)
+   df3.memory_usage()
+
+.. ipython:: python
+
+   df1.A.dtype
+
+.. ipython:: python
+
+   df2.A.dtype
+
+.. ipython:: python
+
+   df3.A.dtype
+
+The data type of all the columns is Object. Using Categorical columns should improve memory usage.
+
+.. ipython:: python
+
+   dfc1 = df1
+   dfc2 = df2
+   dfc1.A = dfc1.A.astype('category')
+   dfc2.A = dfc2.A.astype('category')
+   dfc1.memory_usage()
+
+.. ipython:: python
+
+   dfc2.memory_usage()
+
+.. ipython:: python
+
+   dfc3 = dfc1.append(dfc2)
+   dfc3.memory_usage()
+
+.. ipython:: python
+
```

Comment:
I think you can greatly simplify this entire example and just show the middle section. just show a column as object, then the same one as categorical in 1 shot.

Reply:
Just to make sure I understand... I was trying to show how appending two categorical columns can result in an object column if they don't have the same categories and showing the big difference in storage of an object vs category column. Is that still what we're aiming for?

Reply:
i understand. this example can be pretty short though. you can have 1 column that matches exactly what you are appending and is a category and 1 that is not and turns to object.

Reply:
I see. I'll give it another shot.

Reply:
Thanks for the feedback.